fecgov/fec-cms
# [WIP] Sitemaps

## Summary (required)

- Resolves #5522 

We are transitioning away from search.gov i14Y drawer  to using sitemaps.
- One sitemap for Wagtail pages generated  by Wagtail CMS  named:  `/sitemap-wagtail.xml` and one for non-cms-generated pages named `sitemap-data-legal.xml` that we may manually create/update or find an automated way to create/update it. 
- The sitemaps will be referenced by search.gov via the new `robots_prod.txt` only in production. The other  `robots.txt` tells search engines to Disallow crawling in non-prod  envs (stage. dev, feature).
- The  sitemaps themselves do not need to be conditionally hidden at certain environments because  they always reference `www.fec.gov`, so they would not ever prompt an SE to crawl stage, dev etc, subdomains.
- [ ] **TODO**: See [below comment](https://github.com/fecgov/fec-cms/pull/5579#issuecomment-1414470940) about changes to `wagtail_hooks` and `search_indexing.py`

**Related Issues:** 
Research sitemap generation -  https://github.com/fecgov/fec-cms/issues/5499
Document search.gov sitemap - requirements https://github.com/fecgov/fec-cms/issues/5501
Changes to Policy and Guidance search before transitioning to sitemaps for global search - https://github.com/fecgov/fec-cms/issues/5595)https://github.com/fecgov/fec-cms/issues/5595

### Required reviewers

(Include who is required to review prior to merge. For example: One designer and two front end developer reviews are required prior to merge)

## Impacted areas of the application

General components of the application that this PR will affect:

-  

## Screenshots

(Include a screenshot of the new/updated features in context (“in the wild”). If it is an interface change, include both before and after screenshots)

## Related PRs

Related PRs against other branches:

branch | PR
------ | ------
fix/other_pr | [link]()
feature/other_pr | [link]()

## How to test

-  Test  to `127.0.0.1:8000/robots.txt` with env set to `stage` and  `prod` . You  should see a `disallow` robots.txt in stage, and `sitemap listings` in prod. Also see the conditional statements in  `urls.py` to confirm what  you should see at different envs/URLS.
-  The  sitemaps are not hidden at certain environmemts because  they always reference `www.fec.gov`, so they would not ever prompt an SE to crawl stagge, dev etc, subdomains.
    -127.0.0.1:8000/robots.txt/sitemap-wagtail.xml
     -127.0.0.1:8000/robots.txt/sitemap-data-legal.xml (this is still WIP)
- Run `pytest` . The  new tests confirm the content of the respose at  `/robots.txt` given a certain env (i.e stage, prod)
